### PR TITLE
Initial setup of repository

### PR DESCRIPTION
**Pull request by adam2392**
- Adds all the CI stuff
- Adds the developer workflow copied and pasted
- Changes the name of directory to `pywhy_stats` to keep in line w/ the Python convention that `-` is turned to `_` in package names -> import names

### CODE REVIEW

**Comment by adam2392** [comment]
Someone needs to setup: 

- circleCI
- codecov

since I do not have any permissions unfortunately. @bloebp

**Comment by adam2392** [comment]
This is good to go once CI is green.

@bloebp you can confirm that this is a good copy/paste if CIs pass and if there is no mention of networkx and/or dodisocver and/or pandas within the package. The codecov should fire with 0% coverage once the CIs pass.

Once this is merged, we can activate circleCI for the package.

File: .circleci/config.yml
Comment (bloebp) [inline .circleci/config.yml:62]: Do we necessarily need this part? In particular, do we need graphviz?

File: .github/workflows/main.yml
Comment (bloebp) [inline .github/workflows/main.yml:124]: I saw in dodiscover that 3.8 is required, but I didn't fully get why. Shouldn't all these algorithms work with 3.7 as well?

File: DEVELOPING.md
Comment (bloebp) [inline DEVELOPING.md:18]: I am missing "format" here, i.e., `poetry run poe format`. This is actually all I do normally.

Strictly speaking, this is the only thing that is necessary (assuming that the tests are working).

File: README.md
Comment (bloebp) [inline README.md:9]: What about: "Pywhy-stats serves as Python library for implementations of various statistical methods, such as (un)conditional independence tests, which can be utilized in tasks like causal discovery."

File: README.md
Comment (bloebp) [inline README.md:25]: I guess 3.7 would be a bit more accessible. If not possible, I guess 3.8 is ok as well.

File: README.md
Comment (bloebp) [inline README.md:23]: Don't we also need scikit learn for the k-NN stuff and classifier/regression based methods?

File: doc/api.rst
Comment (bloebp) [inline doc/api.rst:21]: pywhy-stats?

File: doc/api.rst
Comment (bloebp) [inline doc/api.rst:43]: Repetition of the above. What are `Utilities` here?

File: doc/conditional_independence.rst
Comment (bloebp) [inline doc/conditional_independence.rst:7]: References dodiscover

File: pywhy_stats/_version.py
Comment (bloebp) [inline pywhy_stats/_version.py:3]: Do we really need this? I guess this causes issues with older versions, but I also don't see the benefit of it?

File: .circleci/config.yml
Comment (adam2392) [reply .circleci/config.yml:62]: Good catch. Removing.

File: .github/workflows/main.yml
Comment (adam2392) [reply .github/workflows/main.yml:124]: Python 3.7 is reaching EOL this year. Prolly best to not support Python versions that aren't supported by PSF: https://pythonspeed.com/articles/stop-using-python-3.7/

File: README.md
Comment (adam2392) [reply README.md:23]: Since it's not necessary for most tests, we can just keep that optional for now since scikit-learn isn't necessary for things like Chi-square, FisherZ, etc. 

I kept sklearn in the `test` group poetry deps

File: doc/api.rst
Comment (adam2392) [reply doc/api.rst:43]: Removing for now.

File: doc/conditional_independence.rst
Comment (adam2392) [reply doc/conditional_independence.rst:154]: Removed.

File: pyproject.toml
Comment (adam2392) [reply pyproject.toml:9]: Thanks good catch! Moving too fast.. :)

File: pywhy_stats/_version.py
Comment (adam2392) [reply pywhy_stats/_version.py:3]: Honestly... not sure how this works exactly. @darthtrevino added this in dodiscover. My impression is that it allows us to only have version in one place: `pyproject.toml` which makes life easier in terms of making releases(?)

File: DEVELOPING.md
Comment (adam2392) [reply DEVELOPING.md:18]: `poetry run poe format` is the same as `poetry run poe apply_format`. I suppose we should actually consolidate this across all pywhy packages, so the dev exp is as similar as possible

Comment (adam2392) [reply DEVELOPING.md]: Okay changed.

File: doc/conf.py
Comment (bloebp) [inline doc/conf.py:33]: I guess copyright should be pywhy here as well?

**Review by bloebp — APPROVED** [review]

**Pull request merged**
